### PR TITLE
feat(nextjs,edge): Export APIs directly from edge

### PR DESCRIPTION
**Pull request by igneel64**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/types`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [] `npm run build` runs as expected.

- Use `@clerk/edge/vercel-edge` for the edge specific code on `@clerk/nextjs` 
- Ability to export API objects directly in the fashion of `import { users } from '@clerk/nextjs/edge'`

### CODE REVIEW

**Review by nikosdouvlis — APPROVED** [review]
Had a comment about the callback type name, everything else looks great

**Comment by colinclerk** [comment]
Perhaps
WithAuthCallback
WithServerSideAuthCallback
WithEdgeMiddlewareAuthCallback

File: packages/edge/src/vercel-edge/index.ts [hunk 1]
```diff
@@ -4,9 +4,9 @@ import { NextFetchEvent, NextRequest, NextResponse } from 'next/server';
 import { ClerkAPI } from './ClerkAPI';
 import {
   NextMiddlewareResult,
-  WithAuthMiddlewareCallback,
-  WithAuthMiddlewareResult,
-  WithAuthOptions,
+  WithEdgeMiddlewareAuthMiddlewareCallback,
```
[the remark this answers is not in the document]
Comment (igneel64) [reply packages/edge/src/vercel-edge/index.ts:7]: If by accessible we mean, visible yes, if exported from the package it is not.
I would keep it there even if verbose, it makes searching a bit less cumbersome IMO.
No strong opinion though

[the remark this answers is not in the document]
Comment (igneel64) [reply packages/edge/src/vercel-edge/index.ts]: Added as `WithEdgeMiddlewareAuthCallback`

**Pull request merged**